test/mitmproxy/proxy/protocol/test_http2.py

```diff
@@ -862,3 +863,91 @@ def test_connection_terminated(self):
         assert connection_terminated_event.error_code == 5
         assert connection_terminated_event.last_stream_id == 42
         assert connection_terminated_event.additional_data == b'foobar'
+
+
+class TestRequestStreaming(_Http2Test):
+
+    @classmethod
+    def handle_server_event(cls, event, h2_conn, rfile, wfile):
+        if isinstance(event, h2.events.ConnectionTerminated):
+            return False
+        elif isinstance(event, h2.events.RequestReceived):
+            req = rfile.read(109)
```

Comment:
9 bytes headers + 100 bytes bodies
Anything less would throw an error [here](https://github.com/mitmproxy/mitmproxy/pull/2369/files#diff-ae3ce6b83d0e48d6158f2e5fe6583e6bR911) inside `safe_read` because it then reads the length incorrectly